tests/unit/package_managers/test_pip.py

```diff
@@ -3348,14 +3331,17 @@ def test_download_dependencies(
             # Hashes for URL dependencies should be verified no matter what
             verify_checksum_calls = [verify_url_checksum_call]
 
+        if allow_binary:
+            verify_checksum_calls.extend(
+                [
+                    mock.call(w1_path, {ChecksumInfo("sha256", "aaabbb")}),
+                    mock.call(w2_path, {ChecksumInfo("sha256", "cccddd")}),
+                ]
+            )
```

Comment:
do we have any tests for 
- verifying the right checksums (PyPI, user or both)
- not failing when there are no checksums

Reply:
this as well

Reply:
still in progress